### ISSUE TITLE
Parallel VM creation issues and possible improvements

### ISSUE THREAD

**Issue by jay7x** [lima-vm/lima#1354]
### Description

I'd like to start cluster of VMs in parallel as my hardware mostly allows this. But there are few issues I see.

When multiple VMs referring to the same image are creating in parallel they are trying to download the same image simultaneously. First VM downloaded the image will be started while another will proceed with the fallback image. At least that was my experience with 2 VMs.

As I see the only blocking issue here is image downloading. I'd suggest to introduce file locking so all VMs should wait until the image will be downloaded by first started one. I see no other issues on later stages though I might be wrong.

Another issue is `/tmp/lima` mountpoint which is the same for every bundled template.. I'd suggest to move it to `~/.lima/<vm>/tmp/lima` maybe. Otherwise multiple VMs will override the mountpoint and last one wins (as I understand).

**Comment by jay7x** [comment]
More details:

## Test script:
```
#!/bin/bash

VM=test1; limactl start "--name=${VM}" template://ubuntu-lts >${VM} 2>&1 &
VM=test2; limactl start "--name=${VM}" template://ubuntu-lts >${VM} 2>&1 &
VM=test3; limactl start "--name=${VM}" template://ubuntu-lts >${VM} 2>&1 &

wait %1 %2 %3
```

## Expected behavior

3 VMs should be created and running.

## Actual behavior

2 VMs are created, 1 VM failed to be created, 0 VMs are running.

When you start the script all 3 `limactl` processes start downloading the same image from the `ubuntu-lts` template. The VM downloaded it first attempts to start (`test2` in my case). Other 2 VMs tries to download the fallback image. One of those 2 downloaded it first attempts to start (`test3` in my case). Another VM complains it cannot download the image and give up here (`test1` in my case).

None of VMs which have images downloaded were able to start though. `test2` was complaining about IO errors on `vda`. `test3` got kernel panic..

The archive with limactl output, serial logs and VM definitions is attached.
[lima-dbg.tar.gz](https://github.com/lima-vm/lima/files/10697359/lima-dbg.tar.gz)

**Comment by nirs** [comment]
@jay7x Should be fixed in 1.0.0 beta. Please test again.

**Comment by jay7x** [comment]
Ty for confirmation! Feel free to close if it works for you! I'll reopen it if anything.
